Realm/Tests/ArrayPropertyTests.m

```diff
@@ -334,4 +333,105 @@ - (void)testSortByMultipleColumns {
     XCTAssertTrue(checkOrder(@[@"age", @"dogName"], @[@NO, @NO], @[b2, a2, b1, a1]));
 }
 
+- (void)testDeleteLinksAndObjectsInArray
+{
+    RLMRealm *realm = [RLMRealm defaultRealm];
+    [realm beginWriteTransaction];
+
+    EmployeeObject *po1 = [[EmployeeObject alloc] init];
+    po1.age = 40;
+    po1.name = @"Joe";
+    po1.hired = YES;
+
+    EmployeeObject *po2 = [[EmployeeObject alloc] init];
+    po2.age = 30;
+    po2.name = @"John";
+    po2.hired = NO;
+
+    EmployeeObject *po3 = [[EmployeeObject alloc] init];
+    po3.age = 25;
+    po3.name = @"Jill";
+    po3.hired = YES;
+
+    [realm addObject:po1];
+    [realm addObject:po2];
+    [realm addObject:po3];
+
+    CompanyObject *company = [[CompanyObject alloc] init];
+    company.name = @"name";
+    [company.employees addObjects:[EmployeeObject allObjects]];
+    [realm addObject:company];
+
+    [realm commitWriteTransaction];
+
+    RLMArray *peopleInCompany = company.employees;
+
+    // Delete link to employee
+    XCTAssertThrowsSpecificNamed([peopleInCompany removeObjectAtIndex:1], NSException, @"RLMException", @"Not allowed in read transaction");
+    XCTAssertEqual(peopleInCompany.count, 3U, @"No links should have been deleted");
+
+    [realm beginWriteTransaction];
+    XCTAssertThrowsSpecificNamed([peopleInCompany removeObjectAtIndex:3], NSException, @"RLMException", @"Out of bounds");
+    XCTAssertNoThrow([peopleInCompany removeObjectAtIndex:1], @"Should delete link to employee");
+    [realm commitWriteTransaction];
+
+    XCTAssertEqual(peopleInCompany.count, 2U, @"link deleted when accessing via links");
+    EmployeeObject *test = peopleInCompany[0];
+    XCTAssertEqual(test.age, po1.age, @"Should be equal");
+    XCTAssertEqualObjects(test.name, po1.name, @"Should be equal");
+    XCTAssertEqual(test.hired, po1.hired, @"Should be equal");
+    //XCTAssertEqualObjects(test, po1, @"Should be equal"); //FIXME, should work. Asana : https://app.asana.com/0/861870036984/13123030433568
```

Comment:
That Asana task was marked as completed 6 months ago. Does the test still fail? If so, we should open a new GitHub issue.